keyform/static/keyform/js/add_form.js

```diff
@@ -12,4 +12,15 @@ $(document).ready(function() {
             $('#id_keydata_set-TOTAL_FORMS').val(parseInt(form_idx) - 1);
         }
     });
+
+    function mailbox_field() {
+        var key_type_selector = $(this);
+        var pks_with_hide_core_number = key_type_selector.data('pks_with_hide_core_number');
+        pks_with_hide_core_number = String(pks_with_hide_core_number).split(',');
+        var current_key_type = key_type_selector.find('option:selected').attr('value');
+        key_type_selector.closest('.keydata-form').find('[id$=-core_number]').closest('.row').toggle($.inArray(current_key_type, pks_with_hide_core_number) == -1)
+    }
+
+    $('.keydata-form select[data-pks_with_hide_core_number]').on('click change', mailbox_field);
```

Comment:
This event handler doesn't get applied to any KeyData forms that are dynamically added to the page.
Fix it using something like this:
```javascript
$( document ).on( "click", "a.offsite", function() {
  alert( "Goodbye!" );  // jQuery 1.7+
});
```